toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt

```diff
@@ -413,6 +400,132 @@ private fun MapViewEventHandler(
+            },
+            restore = {
+                mutableStateOf(Viewpoint.fromJsonOrNull(it))
+            }
+        )
+    ) {
+        mutableStateOf<Viewpoint?>(null)
+    }
+
+    LaunchedEffect(Unit) {
+        // if there is a persisted viewpoint, restore it when the SceneView enters the composition
+        persistedViewpoint?.let { mapView.setViewpoint(it) }
+        launch {
+            mapView.viewpointChanged.collect {
+                persistedViewpoint = mapView.getViewpointByPersistence(currentViewpointPersistence)
+
+                currentOnViewpointChangedForCenterAndScale?.let { callback ->
+                    val currentViewpoint =
+                        if (persistedViewpoint?.viewpointType != ViewpointType.CenterAndScale) {
+                            mapView.getCurrentViewpoint(ViewpointType.CenterAndScale)
+                        } else persistedViewpoint
+                    currentViewpoint?.let(callback)
+                }
+
+                currentOnViewpointChangedForBoundingGeometry?.let { callback ->
+                    val currentViewpoint =
+                        if (persistedViewpoint?.viewpointType != ViewpointType.BoundingGeometry) {
+                            mapView.getCurrentViewpoint(ViewpointType.BoundingGeometry)
+                        } else persistedViewpoint
+                    currentViewpoint?.let(callback)
+                }
+
+                currentVisibleAreaChanged?.invoke(
+                    mapView.visibleArea
+                        ?: throw IllegalStateException("MapView visible area should not be null")
+                )
+            }
+        }
+        launch {
+            // For performance reasons we use snapshotFlow instead of LaunchedEffect(viewpointPersistence)
+            // here in order to keep track of changes to currentViewpointPersistence at recomposition
+            snapshotFlow { currentViewpointPersistence }
+                .collect {
+                    persistedViewpoint = mapView.getViewpointByPersistence(it)
+                }
+        }
+    }
+}
+
+/**
+ * Normalizes the viewpoint's target geometry.
+ *
+ * If the spatial reference is null or not pannable or geographic, the viewpoint is returned as is.
+ * Otherwise, the viewpoint's target geometry is normalized and a new viewpoint is created with the
+ * normalized geometry.
+ *
+ * @since 200.4.0
+ */
+private fun tryNormalizeViewpoint(viewpoint: Viewpoint): Viewpoint {
+    // Normalization should only be done with pannable or geographic spatial references.
+    val sr = viewpoint.targetGeometry.spatialReference ?: return viewpoint
+    if (!sr.isPannable && !sr.isGeographic) return viewpoint
+    // if normalization fails, just return the viewpoint as is
+    val normalizedGeometry =
+        GeometryEngine.normalizeCentralMeridian(viewpoint.targetGeometry) ?: return viewpoint
+
+    val viewpointToPersist = when (viewpoint.viewpointType) {
```

Comment:
"persist" doesn't make sense in the context of this function, I suggest:
```suggestion
    val normalizedViewpoint = when (viewpoint.viewpointType) {
```